src/Confluent.Kafka/MessageComponentType.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Enumerates different parts of a Kafka message
+    /// </summary>
+    public enum MessageComponentType
```

Comment:
I like your thinking, but I can't think of a name I like.

Reply:
another idea: the serializer classes could be completely decoupled from kafka clients by making `SerializationContext` a dictionary<string,string>, but that would be at the expense of strong typing and (importantly) performance, so a no go. Half way to that is making the enum field a string (shifting the naming problem from the type to a parameter name - less important so easier), which wouldn't have a perf impact in the usual case (only avro, where it won't be significant), but the decoupling isn't complete, so no justification there and strong typing is lost.

Reply:
where I'm at: like the idea of generality, but i don't think it's practical due to perf reasons + naming. This is not a widely touched part of the API, so i think not too important. I did make a minor tweak - shortened the property name to `SerializationContext.Component`.

Reply:
now i see that all the tests failing :-), I remember why I made this a class - because it's nullable. This is convenient for using the serdes outside the context of the producer/consumer because there's often no useful context / this is usually irrelevant to the serde. However, that thinking is backwards. The clients always pass in a non-null reference, so it is in fact desirable that the interface reflects that.